spatialmedia/spherical.py

```diff
@@ -0,0 +1,377 @@
+    return True
+
+
+def ParseSphericalXML(contents, console):
+    """Returns spherical metadata for a set of xml data.
+
+    Args:
+      contents: string, spherical metadata xml contents.
+
+    Returns:
+      dictionary containing the parsed spherical metadata values.
+    """
+    try:
+        parsed_xml = xml.etree.ElementTree.XML(contents)
+    except xml.etree.ElementTree.ParseError:
+        try:
+            index = contents.find("<rdf:SphericalVideo")
+            if index != -1:
+                index += len("<rdf:SphericalVideo")
+                contents = contents[:index] + rdf_prefix + contents[index:]
+            parsed_xml = xml.etree.ElementTree.XML(contents)
+            console("\t\tWarning missing rdf prefix:", rdf_prefix)
+        except xml.etree.ElementTree.ParseError as e:
+            console("\t\tParser Error on XML")
+            console(e)
+            console(contents)
+            return
+
+    sphericalDictionary = dict()
+    for child in parsed_xml.getchildren():
+        if child.tag in SPHERICAL_TAGS.keys():
+            console("\t\tFound: " + SPHERICAL_TAGS[child.tag]
+                    + " = " + child.text)
+            sphericalDictionary[SPHERICAL_TAGS[child.tag]] = child.text
+        else:
+            tag = child.tag
+            if child.tag[:len(spherical_prefix)] == spherical_prefix:
+                tag = child.tag[len(spherical_prefix):]
+            console("\t\tUnknown: " + tag + " = " + child.text)
+
+    return sphericalDictionary
+
+
+def ParseSphericalMpeg4(mpeg4_file, fh, console):
+    """Returns spherical metadata for a loaded mpeg4 file.
+
+    Args:
+      mpeg4_file: mpeg4, loaded mpeg4 file contents.
+      fh: file handle, file handle for uncached file contents.
+
+    Returns:
+      Dictionary stored as (trackName, metadataDictionary)
+    """
+    metadataSets = dict()
+    track_num = 0
+    for element in mpeg4_file.moov_box.contents:
+        if element.name == mpeg.constants.TAG_TRAK:
+            trackName = "Track %d" % track_num
+            console("\t%s" % trackName)
+            track_num += 1
+            for sub_element in element.contents:
+                if sub_element.name == mpeg.constants.TAG_UUID:
+                    if sub_element.contents:
+                        sub_element_id = sub_element.contents[:16]
+                    else:
+                        fh.seek(sub_element.content_start())
+                        sub_element_id = fh.read(16)
+
+                    if sub_element_id == SPHERICAL_UUID_ID:
+                        if sub_element.contents:
+                            contents = sub_element.contents[16:]
+                        else:
+                            contents = fh.read(sub_element.content_size - 16)
+                        metadataSets[trackName] = \
+                            ParseSphericalXML(contents, console)
+    return metadataSets
+
+
+def ParseMpeg4(input_file, console):
+    in_fh = open(input_file, "rb")
```

Comment:
When doing file IO such as this, it's more pythonic to use the with syntax:

with open('filename', 'mode') as f:
  ...

See the bottom of this section for example/rationale:
https://docs.python.org/2/tutorial/inputoutput.html#methods-of-file-objects